src/snowflake/connector/auth.py

```diff
@@ -664,3 +673,23 @@ def build_temporary_credential_name(host, user, cred_type):
     return "{host}:{user}:{driver}:{cred}".format(
         host=host.upper(), user=user.upper(), driver=KEYRING_DRIVER_NAME, cred=cred_type
     )
+
+
+def get_token_from_private_key(
+    user: str, account: str, privatekey_path: str, key_password
+):
+    encoded_password = key_password.encode() if key_password is not None else None
+    with open(privatekey_path, "rb") as key:
+        p_key = load_pem_private_key(
+            key.read(), password=encoded_password, backend=default_backend()
+        )
+
+    private_key = p_key.private_bytes(
+        encoding=Encoding.DER,
+        format=PrivateFormat.PKCS8,
+        encryption_algorithm=NoEncryption(),
+    )
+    auth_instance = AuthByKeyPair(private_key, 1440 * 60)
```

Comment:
The precommit hook called 'black' caused this to fail because I think comments inside the parameter list is not acceptable, but I added a comment describing the behavior above this line.

Reply:
`black` will reformat your files, just run it again with the same commit message, or you can actually manually run `black` with `tox -e fix_lint` once you install `tox` with `python3 -m pip install tox`.

Reply:
changed